src/Engine/ProtoImperative/CodeGen.cs

```diff
@@ -340,7 +340,7 @@ private void EmitPushArrayKey(int symbol, int block, int classIndex)
                         if (procNode != null)
                         {
                             isAccessible = procNode.AccessModifier == ProtoCore.CompilerDefinitions.AccessModifier.Public;
-                            realType = refClassIndex;
+                            realType = refClassIndex = procNode.ClassID;
```

Comment:
The class index needs to be updated here so that it can point to the correct base class for the method call. `refClassIndex` is incorrectly pointing to the derived class on which the method is not defined but inherited.